paddle/majel/allocation.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <memory>
+
+#include <majel/place.h>
+#include "hl_gpu.h"
```

Comment:
Is it a good idea that we create a `paddle/majel/malloc.h`:

```cpp
#pragma once 
#include <majel/place.h>

namespace majel {
namespace malloc {

void init();
void shutdown();

void* malloc(majel::Place place, size_t size);
void free(majel::Place place, void* ptr);
size_t memory_used(majel::Place);

}
}
```

and implement it in `paddle/majel/malloc.cc` by calling `cudaMalloc` directly, so we don't depend on `paddle/cuda`?